spec/lib/extensions/virtual_total_spec.rb

```diff
@@ -1,4 +1,309 @@
 describe VirtualTotal do
+  before(:each) do
+    # rubocop:disable Style/SingleLineMethods, Style/EmptyLineBetweenDefs, Style/AccessorMethodName
+    class VitualTotalTestBase < ActiveRecord::Base
+      self.abstract_class = true
+
+      establish_connection :adapter => 'sqlite3', :database => ':memory:'
+
+      include VirtualFields
+
+      # HACK:  not sure the right way to do this
+      def self.id_increment
+        @id_increment ||= 0
+        @id_increment  += 1
+      end
+    end
+
+    ActiveRecord::Schema.define do
+      def self.connection; VitualTotalTestBase.connection; end
+      def self.set_pk_sequence!(*); end
+      self.verbose = false
+
+      create_table :vt_authors do |t|
+        t.string   :name
+      end
+
+      create_table :vt_books do |t|
+        t.integer  :author_id
+        t.string   :name
+        t.boolean  :published, :default => false
+        t.boolean  :special,   :default => false
+        t.integer  :rating
+        t.datetime :created_on
+      end
+    end
+
+    class VtAuthor < VitualTotalTestBase
+      def self.connection; VitualTotalTestBase.connection; end
```

Comment:
Do you need to override `connection`?

I see it also in `ar_virtual_field` but was never sure if it were needed

Reply:
Honestly, this is a `copy-pasta` of what you did in `spec/lib/exetensions/ar_virtual_spec.rb`... just named things with more human digestible concepts like `book` and `author`, instead of `test_table`.

I tried removing the `set_pk_column` or whatnot, and that was required, and I just assumed (or maybe even tried this as well, but forgot) this was still needed as well.